From 8fcc18acc99776e56472e8e46e364d52488a9cda Mon Sep 17 00:00:00 2001
Message-Id: <8fcc18acc99776e56472e8e46e364d52488a9cda.1441918786.git.yunwei.zhang@intel.com>
In-Reply-To: <8631704be06e00cb7a7a415f13811ca4ece99e8b.1441918786.git.yunwei.zhang@intel.com>
References: <8631704be06e00cb7a7a415f13811ca4ece99e8b.1441918786.git.yunwei.zhang@intel.com>
From: Shubhangi Shrivastava <shubhangi.shrivastava@intel.com>
Date: Tue, 8 Sep 2015 11:23:41 +0530
Subject: [PATCH 05/10] REVERTME [VPG]: drm/i915: Do single audio init

This patch moves the initialization of audio specific data for multiple
encoders into single initialization. This is in continuation of previous
patch to support single audio endpoint for multiple devices.

REVERTME: LP audio is not available upstream and may not be for
some more time which is required for this.

Change-Id: Ia27bca125d20b3e814f4605486080b81e9df54f5
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-10172
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Signed-off-by: Shubhangi Shrivastava <shubhangi.shrivastava@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 14 ++++++++++++++
 drivers/gpu/drm/i915/intel_dp.c      | 27 ---------------------------
 drivers/gpu/drm/i915/intel_hdmi.c    | 22 ----------------------
 3 files changed, 14 insertions(+), 49 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 752df3f..5d0b597 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -13326,7 +13326,21 @@ void set_hdmi_priv(struct drm_device *dev)
 {
 #ifdef CONFIG_EXTCON
 	struct drm_i915_private *dev_priv = dev->dev_private;
+#endif
+	struct hdmi_audio_priv *hdmi_priv;
+
+	hdmi_priv = kzalloc(sizeof(struct hdmi_audio_priv), GFP_KERNEL);
+	if (!hdmi_priv) {
+		pr_err("failed to allocate memory");
+		return;
+	} else {
+		hdmi_priv->dev = dev;
+		hdmi_priv->monitor_type = MONITOR_TYPE_HDMI;
+		hdmi_priv->is_hdcp_supported = true;
+		i915_hdmi_audio_init(hdmi_priv);
+	}
 
+#ifdef CONFIG_EXTCON
 	dev_priv->hotplug_switch.name =
 		kasprintf(GFP_KERNEL, "hdmi_aud");
 #ifdef CONFIG_SUPPORT_LPDMA_HDMI_AUDIO
diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 9c124f9..3d7ed02 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -5392,9 +5392,6 @@ intel_dp_init(struct drm_device *dev, int output_reg, enum port port)
 	struct intel_encoder *intel_encoder;
 	struct drm_encoder *encoder;
 	struct intel_connector *intel_connector;
-#ifdef CONFIG_SUPPORT_LPDMA_HDMI_AUDIO
-	struct hdmi_audio_priv *hdmi_priv;
-#endif
 
 	intel_dig_port = kzalloc(sizeof(*intel_dig_port), GFP_KERNEL);
 	if (!intel_dig_port)
@@ -5449,30 +5446,6 @@ intel_dp_init(struct drm_device *dev, int output_reg, enum port port)
 	intel_encoder->cloneable = 0;
 	intel_encoder->hot_plug = intel_dp_hot_plug;
 
-#ifdef CONFIG_SUPPORT_LPDMA_HDMI_AUDIO
-	hdmi_priv = kzalloc(sizeof(struct hdmi_audio_priv),
-				GFP_KERNEL);
-	if (!hdmi_priv) {
-		pr_err("failed to allocate memory");
-		goto mem_err;
-	}
-	hdmi_priv->dev = dev;
-
-	if (IS_CHERRYVIEW(dev)) {
-		chv_set_lpe_audio_reg_pipe(dev, intel_encoder,
-					hdmi_priv, port);
-	} else {
-		hdmi_priv->hdmi_lpe_audio_reg =
-			I915_HDMI_AUDIO_LPE_B_CONFIG;
-	}
-
-	/* HACK */
-	hdmi_priv->monitor_type = MONITOR_TYPE_HDMI;
-	hdmi_priv->is_hdcp_supported = false;
-	i915_hdmi_audio_init(hdmi_priv);
-#endif
-
-mem_err:
 	intel_connector->panel.fitting_mode = 0;
 	if (!intel_dp_init_connector(intel_dig_port, intel_connector)) {
 		drm_encoder_cleanup(encoder);
diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index 9e5ec94..5e3d9d3 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -2069,9 +2069,6 @@ void intel_hdmi_init(struct drm_device *dev, int hdmi_reg, enum port port)
 	struct intel_digital_port *intel_dig_port;
 	struct intel_encoder *intel_encoder;
 	struct intel_connector *intel_connector;
-#ifdef CONFIG_SUPPORT_LPDMA_HDMI_AUDIO
-	struct hdmi_audio_priv *hdmi_priv;
-#endif
 
 	intel_dig_port = kzalloc(sizeof(*intel_dig_port), GFP_KERNEL);
 	if (!intel_dig_port)
@@ -2130,25 +2127,6 @@ void intel_hdmi_init(struct drm_device *dev, int hdmi_reg, enum port port)
 	intel_dig_port->hdmi.hdmi_reg = hdmi_reg;
 	intel_dig_port->dp.output_reg = 0;
 
-#ifdef CONFIG_SUPPORT_LPDMA_HDMI_AUDIO
-	hdmi_priv = kzalloc(sizeof(struct hdmi_audio_priv), GFP_KERNEL);
-	if (!hdmi_priv) {
-		pr_err("failed to allocate memory");
-	} else {
-		hdmi_priv->dev = dev;
-		hdmi_priv->hdmi_reg = hdmi_reg;
-		if (IS_CHERRYVIEW(dev)) {
-			chv_set_lpe_audio_reg_pipe(dev, intel_encoder,
-						hdmi_priv, port);
-		} else
-			hdmi_priv->hdmi_lpe_audio_reg =
-					I915_HDMI_AUDIO_LPE_B_CONFIG;
-		hdmi_priv->monitor_type = MONITOR_TYPE_HDMI;
-		hdmi_priv->is_hdcp_supported = true;
-		i915_hdmi_audio_init(hdmi_priv);
-	}
-#endif
-
 	intel_hdmi_init_connector(intel_dig_port, intel_connector);
 	intel_connector->panel.fitting_mode = 0;
 }
-- 
1.9.1

